client.ts

```diff
@@ -797,6 +827,7 @@ export class Client {
       objectName,
       partSize,
       metadata: options?.metadata ?? {},
+      headers: options?.headers ?? {},
```

Comment:
```suggestion
      metadata: {...options?.metadata, ...options?.headers},
```
`metadata` is already basically the same thing as headers, so let's just do this and avoid modifying `ObjectUploader`.

Reply:
oh I see now... ok I'll give that whirl a bit later - hopefully I can run the docker image ok - my environment is a little constrained in that sense